docs/reference/modules/discovery/bootstrapping.asciidoc

```diff
@@ -0,0 +1,110 @@
+
+    Sets a list of the <<node.name,node names>> or transport addresses of the
+    initial set of master-eligible nodes in a brand-new cluster. By default
+    this list is empty, meaning that this node expects to join a cluster that
+    has already been bootstrapped.
+
+This setting can be given on the command line when starting up each
+master-eligible node, or added to the `elasticsearch.yml` configuration file on
+those nodes. Once the cluster has formed this setting is no longer required and
+is ignored. It need not be set on master-ineligible nodes, nor on
+master-eligible nodes that are started to join an existing cluster. Note that
+master-eligible nodes should use storage that persists across restarts. If they
+do not, and `cluster.initial_master_nodes` is set, and a full cluster restart
+occurs, then another brand-new cluster will form and this may result in data
+loss.
+
+For a cluster with 3 master-eligible nodes (with <<node.name,node names>>
+`master-a`, `master-b` and `master-c`) the configuration will look as follows:
+
+[source,yaml]
+--------------------------------------------------
+cluster.initial_master_nodes:
+  - master-a
+  - master-b
+  - master-c
+--------------------------------------------------
+
+Alternatively the IP addresses or hostnames (<<node.name,if node name defaults
+to the host name>>) can be used. If there is more than one Elasticsearch node
+with the same IP address or hostname then the transport ports must also be
+given to specify exactly which node is meant:
+
+[source,yaml]
+--------------------------------------------------
+cluster.initial_master_nodes:
+  - 10.0.10.101
+  - 10.0.10.102:9300
+  - 10.0.10.102:9301
+  - master-node-hostname
+--------------------------------------------------
+
+Like all node settings, it is also possible to specify the initial set of
+master nodes on the command-line that is used to start Elasticsearch:
+
+[source,bash]
+--------------------------------------------------
+$ bin/elasticsearch -Ecluster.initial_master_nodes=master-a,master-b,master-c
+--------------------------------------------------
+
+It is technically sufficient to set this on a single master-eligible node in
+the cluster, and only to mention that single node in the setting, but this
+provides no fault tolerance before the cluster has fully formed. It
+is therefore better to bootstrap using at least three master-eligible nodes.
+In any case, when specifying the list of initial master nodes, **it is vitally
+important** to configure each node with exactly the same list of nodes, to
+prevent two independent clusters from forming. Typically you will set this on
+the nodes that are mentioned in the list of initial master nodes.
+
+NOTE: In alpha releases, all listed master-eligible nodes are required to be
+  discovered before bootstrapping can take place. This requirement will be
+  relaxed in production-ready releases.
+
+WARNING: You must put exactly the same set of initial master nodes in each
```

Comment:
This seems like it's restating the "vitally important" info above, so I'd suggest removing the redundancy from one place or the other.